### ISSUE TITLE
Steam Overlay Doesn't Work with NW.js v0.12.2 on Mac OS X

### ISSUE THREAD

**Issue by vaughnroyko** [greenheartgames/greenworks#28]
Through some troubleshooting and discovery, based on https://github.com/greenheartgames/greenworks/issues/16 it appears as though Steam Overlay is completely non-functional in NW.js v0.12.2 on Mac OS X, but works fine in v0.12.1. I have tried all workarounds as described on https://github.com/greenheartgames/greenworks/issues/16 as well.

I believe this is a NW.js specific-issue, as even launching a third-party program using v0.12.2 does not allow the Steam Overlay to pop-up, but in v0.12.1, it does. I posted here as it is the only place where we can maybe get some light shed on the issue. I believe it would be too specific to get any traction over on the NW.js Github.

I took a look at the changelog from 0.12.1 to 0.12.2 and I can't find anything that could of broke the functionality/rendering:
https://github.com/nwjs/nw.js/blob/master/CHANGELOG.md
https://github.com/nwjs/nw.js/commits/nw13

Perhaps it was some dependencies? No idea.

As a side note, Steam Overlay doesn't work on any Electron version that I have tried on Mac OS X. Perhaps I didn't try to go back far enough though. I assume there would be a similar "breaking" point or version, possibly back in it's atom-shell days.

**Comment by morgondag** [comment]
Can confirm that steam overlay dosen't work for NW.js v0.12.2
is v0.12.1 available for download anywhere?

**Comment by hokein** [comment]
@vaughnroyko 
Can you provide a worked sample of nw.js v0.12.1 here?

**Comment by vaughnroyko** [comment]
Here is 0.12.1 for download: https://groups.google.com/forum/#!msg/nwjs-general/bvdnDJAx91U/0womlMli4FUJ

As for my test set-up, i'm simply using the demo/quick start files as shown on the main NW.js Github: https://github.com/nwjs/nw.js#quick-start with force refresh edits here: https://github.com/greenheartgames/greenworks/wiki/Troubleshooting#steam-overlay-is-unresponsive--frozen

**Comment by hokein** [comment]
@vaughnroyko 
I have the same result with you. Only NW.js 0.12.1 works, neithor NW.js 0.12.2 or Electron(v0.29.2) works. See my comment in #16 .

I haven't much idea on why it doesn't works on NW.js v0.12.2 and Electron, but I think it's more likely to NW.js and Electron rather than Greenworks.

**Comment by vaughnroyko** [comment]
Yep, I assumed it was specifically an NW.js/Electron issue and not Greenworks. I was hoping somebody here would be able to figure out why. As I said, I can't see any changes that would break this in the changelog/commits for 0.12.1 to 0.12.2. I have also tried the new alpha of NW.js and it has the same issue, so if we can't figure it out, Greenworks is basically stuck on NW.js 0.12.1.

**Comment by hokein** [comment]
> As I said, I can't see any changes that would break this in the changelog/commits for 0.12.1 to 0.12.2. 

Besides the commits in https://github.com/nwjs/nw.js/commits/nw12, there are also changes in the https://github.com/nwjs/chromium.src/tree/nw12, I could not figure out which commits is using in v0.12.1 and v0.12.2.

Also have test on Electron v0.22.3, which is using the same Chromium version 41 with NW.js v0.12.1, unfortunately, the overlay still doesn't work.

**Comment by morgondag** [comment]
Just to clarify:
0.12.1 works! 
But **NOT** with ```--in-process-gpu```in your package.json this will result in a white-screen.
use the forceRefreshCanvas for any other problem.
Or if you are making a canvas app you will be fine without it if it runs constantly.

Windows works with the newest 0.12.2.
I haven't found out what drastically changed in the newer release.
Main point to upgrade to newer version is the poor CSS animation performance on mac.

Not sure about Linux yet will make builds for that later on.

**Comment by hokein** [comment]
>Just to clarify:
0.12.1 works! 
But NOT with --in-process-gpuin your package.json this will result in a white-screen.

I also met it. 

>Windows works with the newest 0.12.2.

Glad to see it.
So far, we can confirm that  OS X doesn't work and Windows does with v 0.12.2.

**Comment by remvst** [comment]
Is there anything I can do to get the overlay working? I'm using NW.js 0.22.3 and the overlay still doesn't seem to show up on Mac (I have yet to try on Windows)

**Comment by mrdev023** [comment]
I have same problem on 0.26.6
